docs/topics/gradle-best-practices.md

```diff
@@ -0,0 +1,141 @@
+
+Although the initial setup may be time-consuming, you will find it easy to maintain and add new build logic once you complete it.
+
+Learn more in Gradle's documentation about [Convention plugins](https://docs.gradle.org/current/userguide/custom_plugins.html#sec:convention_plugins).
+
+## Optimize
+
+This section provides strategies to enhance the performance and efficiency of your Gradle builds.
+
+### Use local build cache
+
+Use a local build cache to save time by reusing outputs produced by other builds. The build cache can retrieve outputs from 
+any earlier build that you have already created.
+
+Learn more in Gradle's documentation about their [Build cache](https://docs.gradle.org/current/userguide/build_cache.html).
+
+### Use configuration cache
+
+> The configuration cache doesn't support all core Gradle plugins yet. For the latest information, see Gradle's
+> [table of supported plugins](https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:plugins:core).
+>
+{style="note"}
+
+Use the configuration cache to significantly improve build performance by caching the result of the configuration phase
+and reuse it for subsequent builds. If nothing changes in a build file (or its dependencies) the build file isn't
+recompiled.
+
+Learn more in Gradle's documentation about their [Configuration cache](https://docs.gradle.org/current/userguide/configuration_cache.html).
+
+### Improve build times for multiple targets
+
+When your multiplatform project includes multiple targets, tasks like `build` and `assemble` can compile the same code 
+multiple times for each target, leading to longer compilation times.
+
+If you're actively developing and testing a specific platform, run the corresponding `linkDebug*` task instead.
+
+For more information, see [Tips for improving compilation time](https://kotlinlang.org/docs/native-improving-compilation-time.md#gradle-configuration).
+
+### Migrate from kapt to KSP
+
+If you're using a library that relies on the [kapt](kapt.md) compiler plugin, check whether you can switch to using the [Kotlin Symbol Processing (KSP) API](https://kotlinlang.org/docs/ksp-overview.html)
+instead. The KSP API improves build performance by reducing annotation processing time. KSP is faster and more efficient
+than kapt, as it processes source code directly without generating intermediary Java stubs.
+
+To learn more about how KSP compares to kapt, check out [why KSP](ksp-why-ksp.md).
+
+### Use modularization
+
+<primary-label ref="advanced"/>
+
+> Modularization only benefits projects of moderate to large size. It doesn't provide advantages for projects based
+> on a microservices architecture.
+>
+{style="note"}
+
+Use a modularized project structure to accelerate build speed and enable easier parallel development. Structure your
+project into one root project and one or more subprojects. If there are changes in only one of the subprojects, Gradle
+rebuilds only that specific subproject.
```

Comment:
```suggestion
project into one root project and one or more subprojects. If changes only affect one of the subprojects, Gradle
rebuilds only that specific subproject.
```

Minor rewording, because if `app subproject` depends on `lib subproject`, any changes in `lib subproject` will cause a rebuild of `app subproject`.